src/app/routes/applications/new-role/new-role.component.ts

```diff
@@ -393,6 +394,16 @@ export class NewRoleComponent implements OnInit, AfterViewInit, OnDestroy {
     }
   }
 
+  updateField() {
+    if (this.fieldsForm.valid) {
+      this.dataSource.data.splice(this.fieldIndex, 1, this._extractValidationObject(this.fieldsForm.value))
+      this.fieldsForm.reset();
+      this.showFieldsForm = false;
```

Comment:
It's more like a question, but is it possible to merge `showFieldsForm` and `isEditFieldForm` into one property? Do you know that?
I'm asking because you are setting both with the same values :)

Reply:
So `showFieldsForm` is a property that shows or hides the form and `isEditFieldForm` is what toggles the button from save to update

Reply:
Ok, thanks 👍